gorm/transaction.go

```diff
@@ -39,13 +43,33 @@ func FromContext(ctx context.Context) (txn *Transaction, ok bool) {
 	return
 }
 
+// GetReadOnlyDB returns the read only db instance stored in the ctx if there is no txn in use with read/write database
+func GetReadOnlyDB(ctx context.Context) (*gorm.DB, error) {
+	logger := ctxlogrus.Extract(ctx)
+	txn, ok := FromContext(ctx)
+	if !ok {
+		return nil, ErrCtxTxnMissing
+	}
+	if txn.current != nil {
+		logger.Warnf("GetReadOnlyDB: Txn already initialized with read/write DB")
```

Comment:
1. Maybe the caller would like to know that it asked for a read-only transaction and got a read-write one instead.
2. Extracting the logger from the context assumes this code is used with the gRPC gateway which might not be the case.